### PR TITLE
feat(icon): Add icon's storybook

### LINKED ISSUE #234: missing Icon storybook
Icon has no storybook, please add a few examples - thanks

### PR DESCRIPTION
**Pull request by laviro**
fix #234: Add storybook for PatternFly Icon and Font-Awesome Icon. 

**Link to Storybook**: https://github.com/laviro/patternfly-react

**Additional issues**:

### CODE REVIEW

**Comment by sharvit** [comment]
@laviro thank you for adding this storybook and welcome to patternfly-react.

1. The storybook link isn't working (at least for me), can you fix it please?
2. Look like you have some linting errors
    You can run `npm run lint` to run the linting script and see the errors
    You can run `npm run lint -- --fix` to let the linting script to try and fix the errors automatically

**Comment by laviro** [comment]
@sharvit thank you for the quick reply !
I will fix this problems and upload a new version soon :)

**Comment by laviro** [comment]
You can check the Icon's storybook on this link:

https://laviro.github.io/patternfly-react/?selectedKind=Icon&selectedStory=Patternfly%20Icons&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs

**Comment by ohadlevy** [comment]
thanks, anything else that is missing before this can be merged?

File: src/components/Icon/Icon.stories.js [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+IconStories.addDecorator(withKnobs);
+IconStories.addDecorator(
+  defaultTemplate({
+    title: 'Icons',
+    documentationLink: `${DOCUMENTATION_URL.PATTERNFLY_ORG_WIDGETS}#Icons`,
+    reactBootstrapDocumentationLink: `${
+      DOCUMENTATION_URL.REACT_BOOTSTRAP_COMPONENT
+    }icon/`
+  })
```
Comment (sharvit) [inline src/components/Icon/Icon.stories.js]: Both of the links here didn't actually work for me.

**Comment by laviro** [comment]
@sharvit I have updated the documentation's url,
hope that everything is good now.

**Comment by jeff-phillips-18** [comment]
Could you update the comments at https://github.com/patternfly/patternfly-react/blob/master/src/components/Icon/Icon.js#L14 so they are more accurate in the storybook?

type should list the possibilities ('fa' or 'pf')
name should specify that it could be the pf name or fa name without the prefix.

**Comment by laviro** [comment]
@jeff-phillips-18 I have updated the comments :)

**Comment by jeff-phillips-18** [comment]
@sharvit Any further comments?

File: storybook/constants.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ export const BASE_URL = {
 export const DOCUMENTATION_URL = {
   PATTERNFLY_ORG_COMMUNICATION: `${BASE_URL.PATTERNFLY_ORG}communication/`,
   PATTERNFLY_ORG_WIDGETS: `${BASE_URL.PATTERNFLY_ORG}widgets/`,
+  PATTERNFLY_ORG_STYLES: `https://www.patternfly.org/styles/`,
```
Comment (sharvit) [inline storybook/constants.js]: Shouldn't it be?
```jsx
PATTERNFLY_ORG_STYLES: `${BASE_URL.PATTERNFLY_ORG}styles/`,
```

Comment (laviro) [reply storybook/constants.js]: @sharvit I agree, but the 
`BASE_URL.PATTERNFLY_ORG`
is equal to 
'http://www.patternfly.org/pattern-library/'
and I didn't want to brake the code.

Should we add another constant called 'PATTERN_LIBRARY' ? 
and set the `PATTERNFLY_ORG` to be just 'http://www.patternfly.org' ?

Comment (jeff-phillips-18) [reply storybook/constants.js]: Would make sense to have

PATTERNFLY_ORG = 'http://www.patternfly.org'
PATTERNFLY_LIBRARY = PATTERNFLY_ORG/pattern-library/
PATTERNFLY_STYLES = PATTERNFLY_ORG/styles/

Though we would then need to update all current uses of PATTERNFLY_ORG  😢

Comment (priley86) [reply storybook/constants.js]: whoops!! apologies... well my editor has global find/replace, does yours? Feel free to do this...

Comment (priley86) [reply storybook/constants.js]: oh - so it looks like `PATTERNFLY_ORG` was not actually being used everywhere and this is OK now...phew!

**Pull request merged**